src/Servers/HttpSys/src/RequestProcessing/Response.cs

```diff
@@ -527,128 +516,92 @@ private static bool Matches(string knownValue, StringValues input)
             // else known single-value header.
         }
 
-        try
+        fixed (HttpApiTypes.HTTP_KNOWN_HEADER* pKnownHeaders = &_nativeResponse.Response_V1.Headers.KnownHeaders)
```

Comment:
@jkotas Would it be possible to replace this with ref locals?

Reply:
Possible - yes. Beneficial - depends on your style preference, the short-term pinning is not a performance concern.

Reply:
I just experimented with it, so it's purely a style difference. Unsafe pointers vs managed pointers.